virtcontainers/clh.go

```diff
@@ -1202,141 +1175,195 @@ func (o *CLICmdline) Build(cmdline *CommandLine) {
-	d.vsock = vsock
-}
+func openAPIClientError(err error) error {
 
-func (d *DefaultCLIBuilder) SetAPISocket(apiSocket *CLIAPISocket) {
-	d.apiSocket = apiSocket
-}
+	if err == nil {
+		return nil
+	}
+
+	reason := ""
+	if apierr, ok := err.(chclient.GenericOpenAPIError); ok {
+		reason = string(apierr.Body())
+	}
 
-func (d *DefaultCLIBuilder) SetLogFile(logFile *CLILogFile) {
-	d.logFile = logFile
+	return fmt.Errorf("error: %v reason: %s", err, reason)
 }
 
-func (d *DefaultCLIBuilder) GetCommandLine() (*CommandLine, error) {
+func (clh *cloudHypervisor) bootVM(ctx context.Context) error {
 
-	cmdLine := &CommandLine{}
+	cl := clh.client()
 
-	if d.serial != nil {
-		d.serial.Build(cmdLine)
+	if clh.config.Debug {
+		bodyBuf, err := json.Marshal(clh.vmconfig)
+		if err != nil {
+			return err
+		}
+		clh.Logger().WithField("body", string(bodyBuf)).Debug("VM config")
 	}
+	_, err := cl.CreateVM(ctx, clh.vmconfig)
 
-	if d.console != nil {
-		d.console.Build(cmdLine)
+	if err != nil {
+		return openAPIClientError(err)
 	}
 
-	if d.apiSocket != nil {
-		d.apiSocket.Build(cmdLine)
-	}
+	info, _, err := cl.VmInfoGet(ctx)
 
-	if d.logFile != nil {
-		d.logFile.Build(cmdLine)
+	if err != nil {
+		return openAPIClientError(err)
 	}
 
-	if d.cpus != nil {
-		d.cpus.Build(cmdLine)
-	}
-	if d.memory != nil {
-		d.memory.Build(cmdLine)
-	}
-	if d.disk != nil {
-		d.disk.Build(cmdLine)
-	}
-	if d.rng != nil {
-		d.rng.Build(cmdLine)
+	clh.Logger().Debugf("VM state after create: %#v", info)
+
+	if info.State != "Created" {
+		return fmt.Errorf("VM state is not 'Created' after 'CreateVM'")
 	}
-	if d.vsock != nil {
-		d.vsock.Build(cmdLine)
+
+	clh.Logger().Debug("Booting VM")
+	_, err = cl.BootVM(ctx)
+
+	if err != nil {
+		return openAPIClientError(err)
 	}
-	if d.fs != nil {
-		d.fs.Build(cmdLine)
+
+	info, _, err = cl.VmInfoGet(ctx)
+
+	if err != nil {
+		return openAPIClientError(err)
 	}
-	if d.kernel != nil {
-		d.kernel.Build(cmdLine)
+
+	clh.Logger().Debugf("VM state after boot: %#v", info)
+
+	if info.State != "Running" {
+		return fmt.Errorf("VM state is not 'Running' after 'BootVM'")
 	}
-	if d.nets != nil {
-		d.nets.Build(cmdLine)
+
+	return nil
+}
+
+func (clh *cloudHypervisor) addVSock(cid int64, path string) {
+	clh.Logger().WithFields(log.Fields{
+		"path": path,
+		"cid":  cid,
+	}).Info("Adding HybridVSock")
+
+	clh.vmconfig.Vsock = append(clh.vmconfig.Vsock, chclient.VsockConfig{Cid: cid, Sock: path})
+}
+
+func (clh *cloudHypervisor) addNet(e Endpoint) {
+	clh.Logger().WithField("endpoint-type", e).Debugf("Adding Endpoint of type %v", e)
+	mac := e.HardwareAddr()
+	tapPath := e.NetworkPair().TapInterface.TAPIface.Name
+	clh.Logger().WithFields(log.Fields{
+		"mac": mac,
+		"tap": tapPath,
+	}).Info("Adding Net")
+
+	// FIXME: This is required by CH
+	// remove after PR is merged:
+	// https://github.com/cloud-hypervisor/cloud-hypervisor/pull/480
+	ip := "0.0.0.0"
+	mask := "0.0.0.0"
```

Comment:
@jcvenegas @likebreath I've just submitted https://github.com/cloud-hypervisor/cloud-hypervisor/pull/516 which should solve this issue. Let me know if that works for you.

Reply:
#516 has been merged.
Can the above be removed ?